.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,160 @@
+                        deployDir = 'moe:' + deployDir;
+                    } else {
+                        deployDir = 'chalmers:' + deployDir;
+                    }
+                    if (env.BRANCH_NAME =~ '^(hotfix|release)-.*') {
+                        deployDir += '-staging';
+                    }
+                    if (env.BRANCH_NAME != 'master') {
+                        cliDir += '-' + gemmaVersion;
+                    }
+                }
+            }
+        }
+        stage('Build') {
+            steps {
+                sh 'mvn -B package -DskipTests'
+            }
+        }
+        stage('Run quick unit tests') {
+            steps {
+                sh 'mvn -B test -DexcludedGroups=SlowTest'
+            }
+        }
+        stage('Run slow unit tests and integration tests') {
+            when {
+                branch 'hotfix-*';
+                branch 'release-*';
+                branch 'development'
+            }
+            steps {
+                sh 'mvn -B test -Dgroups=SlowTest'
+                lock('test-database/' + testdbName) {
+                    sh 'mvn -B verify -DskipUnitTests'
+                }
+            }
+        }
+        stage('Deploy artifacts') {
+            when {
+                branch 'master';
+                branch 'development'
+            }
+            steps {
+                sh 'mvn -B deploy -DskipTests'
+            }
+        }
+        stage('Deploy Maven website') {
+            when {
+                branch 'master';
+                branch 'hotfix-*';
+                branch 'release-*';
+                branch 'development'
+            }
+            steps {
+                sh 'mvn -B site-deploy';
+            }
+        }
+        stage ('Deploy WAR') {
+            when {
+                branch 'master'
+                branch 'hotfix-*';
+                branch 'release-*'
+                branch 'development'
+            }
+            steps {
+                sh 'rsync gemma-web/src/main/config/log4j.properties ${deployDir}/lib/log4j.properties';
+                sh 'rsync gemma-web/target/Gemma.war ${deployDir}/Gemma.war';
+                sh 'ln -sf ${siteDir}/gemma/gemma-${gemmaVersion} ${dataDir}/gemma-devsite'
+                sh 'ln -sf ${siteDir}/baseCode/baseCode-${baseCodeVersion} ${dataDir}/basecode-site';
+            }
+        }
+        stage ('Deploy CLI') {
+            when {
+                branch 'master'
+                branch 'hotfix-*';
+                branch 'release-*'
+            }
+            steps {
+                sh 'rsync -adv gemma-cli/target/appassembler ${cliDir}'
+            }
+        }
+    }
+    post {
+        always {
+            archiveArtifacts artifacts: '**/target/*.jar, **/target/*.war', fingerprint: true
+            junit testResults: '**/target/surefire-reports/*.xml', allowEmptyResults: true
+            junit testResults: '**/target/failsafe-reports/*.xml', allowEmptyResults: true
```

Comment:
Individual stages can have a post, so this should be moved in the testing stages.